autotest/utilities/test_gdalwarp.py

```diff
@@ -1113,6 +1118,37 @@ def test_gdalwarp_40(gdalwarp_path, tmp_path):
     assert ds.GetRasterBand(1).Checksum() == cs_ov0
     ds = None
 
+    # Should select overview 0
+    gdaltest.runexternal(f"{gdalwarp_path} {src_tif} {dst_tif} -overwrite -ovr 0")
+
+    ds = gdal.Open(dst_tif)
+    assert ds.GetRasterBand(1).Checksum() == cs_ov0
+    ds = None
+
+    # Should select overview 0 (no overwrite)
+    gdaltest.runexternal(f"{gdalwarp_path} {src_tif} {dst_tif} -ovr 0")
+
+    # Repeat with no output file and no overwrite (takes a different code path)
+    os.unlink(dst_tif)
+    gdaltest.runexternal(f"{gdalwarp_path} {src_tif} {dst_tif} -ovr 0")
+
+    # Repeat with wrong resolution output file and no overwrite (takes a different code path)
+    os.unlink(dst_tif)
+    # Crates a full res output file
+    gdaltest.runexternal(f"{gdalwarp_path} {src_tif} {dst_tif}")
+    gdaltest.runexternal(f"{gdalwarp_path} {src_tif} {dst_tif} -ovr 0")
+    ds = gdal.Open(dst_tif)
+    # FIXME: This fails: is that expected?
+    # assert ds.GetRasterBand(1).Checksum() == cs_ov0
```

Comment:
yes, that's expected. You are updating an existing dataset, created at full resolution and size, with a overview of a source dataset. Hence the checksum of the output cannot match the source overview, because there is not the same number of pixels. I believe lines 1128 to 1143 don't bring much in terms o testing coverage and could probably be removed

Reply:
do you mind if I keep (most of) them ?  They have been helpful while developing, because they trigger different code paths (depending on whether there is or isn't an existing output file).

Reply:
fine for me